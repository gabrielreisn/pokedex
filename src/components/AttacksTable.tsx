import { TypeTag } from "./TypeTag";
import { AttacksTableProps, AttackType } from "./types";

export const AttacksTable = ({ title, attacks }: AttacksTableProps) => (
  <>
    <span className='text-left ml-2 text-lg font-semibold'>{title}</span>
    <table className='my-3 mx-1'>
      <thead>
        <tr>
          <th className='text-left'>Attack</th>
          <th>Type</th>
          <th>Power</th>
        </tr>
      </thead>
      <tbody className='text-xs'>
        {attacks.map((el: AttackType) => (
          <tr key={`${el.name} - ${el.damage}`}>
            <td className='text-left'>{el.name}</td>
            <td>
              <TypeTag value={el.type} />
            </td>
            <td>{el.damage}</td>
          </tr>
        ))}
      </tbody>
    </table>
  </>
);
