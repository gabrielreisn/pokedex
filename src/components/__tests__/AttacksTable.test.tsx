import React from "react";
import { render } from "@testing-library/react";
import { AttacksTable } from "../AttacksTable";

const props = {
  attacks: [
    {
      damage: 12,
      name: "fast-attack-1",
      type: "Normal"
    },
    {
      damage: 123,
      name: "fast-attack-2",
      type: "Normal"
    },
    {
      damage: 1234,
      name: "fast-attack-3",
      type: "Normal"
    }
  ],
  title: "Fast Attacks"
};

describe("<AttacksTable />", () => {
  it("matches the snapshot", () => {
    const wrapper = render(<AttacksTable {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("has two normal type tag attacks", () => {
    const { getAllByText } = render(<AttacksTable {...props} />);
    const normalTypeAttack = getAllByText("Normal");
    expect(normalTypeAttack).toHaveLength(3);
  });
});
