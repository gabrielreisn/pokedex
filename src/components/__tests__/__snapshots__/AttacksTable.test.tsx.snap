// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttacksTable /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="text-left ml-2 text-lg font-semibold"
      >
        Fast Attacks
      </span>
      <table
        class="my-3 mx-1"
      >
        <thead>
          <tr>
            <th
              class="text-left"
            >
              Attack
            </th>
            <th>
              Type
            </th>
            <th>
              Power
            </th>
          </tr>
        </thead>
        <tbody
          class="text-xs"
        >
          <tr>
            <td
              class="text-left"
            >
              fast-attack-1
            </td>
            <td>
              <span
                class="rounded"
                style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
              >
                Normal
              </span>
            </td>
            <td>
              12
            </td>
          </tr>
          <tr>
            <td
              class="text-left"
            >
              fast-attack-2
            </td>
            <td>
              <span
                class="rounded"
                style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
              >
                Normal
              </span>
            </td>
            <td>
              123
            </td>
          </tr>
          <tr>
            <td
              class="text-left"
            >
              fast-attack-3
            </td>
            <td>
              <span
                class="rounded"
                style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
              >
                Normal
              </span>
            </td>
            <td>
              1234
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <span
      class="text-left ml-2 text-lg font-semibold"
    >
      Fast Attacks
    </span>
    <table
      class="my-3 mx-1"
    >
      <thead>
        <tr>
          <th
            class="text-left"
          >
            Attack
          </th>
          <th>
            Type
          </th>
          <th>
            Power
          </th>
        </tr>
      </thead>
      <tbody
        class="text-xs"
      >
        <tr>
          <td
            class="text-left"
          >
            fast-attack-1
          </td>
          <td>
            <span
              class="rounded"
              style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
            >
              Normal
            </span>
          </td>
          <td>
            12
          </td>
        </tr>
        <tr>
          <td
            class="text-left"
          >
            fast-attack-2
          </td>
          <td>
            <span
              class="rounded"
              style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
            >
              Normal
            </span>
          </td>
          <td>
            123
          </td>
        </tr>
        <tr>
          <td
            class="text-left"
          >
            fast-attack-3
          </td>
          <td>
            <span
              class="rounded"
              style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
            >
              Normal
            </span>
          </td>
          <td>
            1234
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
